packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -224,6 +240,105 @@ export class RunningSummarizer implements IDisposable {
+		let refSequenceNumber = -1;
+		// In case we haven't received the lastestAck yet, just return.
+		if (lastAck !== undefined) {
+			refSequenceNumber = lastAck.summaryOp.referenceSequenceNumber;
+			const summaryLogger = this.tryGetCorrelatedLogger(refSequenceNumber) ?? this.mc.logger;
+			try {
+				const summaryOpHandle = lastAck.summaryOp.contents.handle;
+				const summaryAckHandle = lastAck.summaryAck.contents.handle;
+				while (this.summarizingLock !== undefined) {
+					summaryLogger.sendTelemetryEvent({
+						eventName: "RefreshAttemptWithSummarizerRunning",
+						referenceSequenceNumber: refSequenceNumber,
+						proposalHandle: summaryOpHandle,
+						ackHandle: summaryAckHandle,
+					});
+					await this.summarizingLock;
+				}
+
+				// Make sure we block any summarizer from being executed/enqueued while
+				// executing the refreshLatestSummaryAck.
+				// https://dev.azure.com/fluidframework/internal/_workitems/edit/779
+				await this.lockedSummaryAction(
+					() => {},
+					async () =>
+						this.refreshLatestSummaryAckCallback({
+							proposalHandle: summaryOpHandle,
+							ackHandle: summaryAckHandle,
+							summaryRefSeq: refSequenceNumber,
+							summaryLogger,
+						}).catch(async (error) => {
+							// If the error is 404, so maybe the fetched version no longer exists on server. We just
+							// ignore this error in that case, as that means we will have another summaryAck for the
+							// latest version with which we will refresh the state. However in case of single commit
+							// summary, we might me missing a summary ack, so in that case we are still fine as the
+							// code in `submitSummary` function in container runtime, will refresh the latest state
+							// by calling `refreshLatestSummaryAckFromServer` and we will be fine.
+							if (
+								isFluidError(error) &&
+								error.errorType === DriverErrorType.fileNotFoundOrAccessDeniedError
+							) {
+								summaryLogger.sendTelemetryEvent(
+									{
+										eventName: "HandleSummaryAckErrorIgnored",
+										referenceSequenceNumber: refSequenceNumber,
+										proposalHandle: summaryOpHandle,
+										ackHandle: summaryAckHandle,
+									},
+									error,
+								);
+							} else {
+								throw error;
+							}
+						}),
+					() => {},
+				);
+			} catch (error) {
+				summaryLogger.sendErrorEvent(
+					{
+						eventName: "HandleLastSummaryAckError",
+						referenceSequenceNumber: refSequenceNumber,
+						handle: lastAck?.summaryOp?.contents?.handle,
+						ackHandle: lastAck?.summaryAck?.contents?.handle,
+					},
+					error,
+				);
+			}
+			refSequenceNumber++;
+		}
+		return refSequenceNumber;
+	}
+
+	/**
+	 * Responsible for receiving and processing all the summaryAcks.
+	 * @param lastAckRefNumber - Before initializing the summarization, if there is a last Ack to be processed, use it.
+	 */
+	private async handleSummaryAcks(lastAckRefNumber: number) {
```

Comment:
We should probably name this that reflects it will continuously wait for acks. Also, it sounds very similar to handleSummaryAck :-).
Maybe something like `processIncomingSummaryAcks`?

Reply:
Better indeed.